jgrapht-core/src/main/java/org/jgrapht/alg/tour/PalmerHamiltonianCycle.java

```diff
@@ -0,0 +1,152 @@
+/*
+ * (C) Copyright 2018-2018, by Alexandru Valeanu and Contributors.
+ *
+ * JGraphT : a free Java graph-theory library
+ *
+ * This program and the accompanying materials are dual-licensed under
+ * either
+ *
+ * (a) the terms of the GNU Lesser General Public License version 2.1
+ * as published by the Free Software Foundation, or (at your option) any
+ * later version.
+ *
+ * or (per the licensee's choosing)
+ *
+ * (b) the terms of the Eclipse Public License v1.0 as published by
+ * the Eclipse Foundation.
+ */
+package org.jgrapht.alg.tour;
+
+import org.jgrapht.Graph;
+import org.jgrapht.GraphPath;
+import org.jgrapht.GraphTests;
+import org.jgrapht.graph.GraphWalk;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Palmer's algorithm for computing Hamiltonian cycles in graphs that meet Ore's condition.
+ *
+ * <p>
+ *     A Hamiltonian cycle, also called a Hamiltonian circuit, Hamilton cycle, or Hamilton circuit, is a graph cycle
+ *     (i.e., closed loop) through a graph that visits each node exactly once (Skiena 1990, p. 196).
+ * </p>
+ *
+ * <p>
+ *     This is an implementation of the algorithm described by E. M. Palmer in his paper. The algorithm takes a simple
+ *     graph that meets Ore's condition (see {@link GraphTests#hasOreProperty(Graph)}) and returns a Hamiltonian cycle.
+ *     The algorithm runs in $O(|V|^2)$ and uses $O(|V|)$ space.
+ * </p>
+ *
+ * <p>
+ *    See <a href="https://en.wikipedia.org/wiki/Ore%27s_theorem">wikipedia</a> for a short description of Ore's theorem
+ *    and the resulting algorithm and <a href="https://www.sciencedirect.com/science/article/pii/S0898122197002253">Palmer's paper</a> for more details.
```

Comment:
please use proper citation. If this link changes, we no longer know which paper you were referencing.

Reply:
Done.